diff --git a/components/form/base/form-base-element.d.ts b/components/form/base/form-base-element.d.ts
index e52556265a0af2215aef722b3c2abdcf548c8e00..3a3c839158e9c18760e5ac1eee8f7e3facadac02 100644
--- a/components/form/base/form-base-element.d.ts
+++ b/components/form/base/form-base-element.d.ts
@@ -7,6 +7,7 @@ export declare class FormBaseElement extends LitElement {
     autofocus: boolean;
     light: boolean;
     novalidation: boolean;
+    valid: boolean;
     protected ariaController: AriaController;
     protected disabledController: DisabledController;
     protected errorController: ErrorController;